static/js/util.js

```diff
@@ -284,44 +293,69 @@ export function clean_user_content_links(html) {
         } else {
             elt.removeAttribute("target");
         }
+    }
+    return content.innerHTML;
+}
 
-        const is_inline_image =
-            elt.parentElement && elt.parentElement.classList.contains("message_inline_image");
-        if (is_inline_image) {
-            // For inline images we want to handle the tooltips explicitly, and disable
-            // the browser's built in handling of the title attribute.
-            if (elt.getAttribute("title")) {
-                elt.setAttribute("aria-label", elt.getAttribute("title"));
-                elt.removeAttribute("title");
-            }
+export function initialize() {
+    $("#main_div").on("mouseenter", "a", (e) => {
+        let elt = e.target;
+        if (elt.nodeName === "IMG") {
+            elt = elt.parentElement;
+        }
+        const href = elt.getAttribute("href");
+        const title = elt.getAttribute("title");
+        const aria_label = elt.getAttribute("aria-label");
+        const parentElement = elt.parentElement;
+        if (
+            !href ||
+            (title && !parentElement.classList.contains("message_inline_image")) ||
+            aria_label ||
+            elt.classList.contains("previewable")
+        ) {
+            return;
+        }
+        if (is_url_previewable(href)) {
+            elt.classList.add("previewable");
```

Comment:
Could you do a `return` here rather than intending all the code below via putting it in a giant `else` clause?